mkosi/state.py

```diff
@@ -21,6 +21,16 @@ def __init__(self, args: MkosiArgs, config: MkosiConfig, workspace: Path) -> Non
         self.pkgmngr.mkdir()
         self.install_dir.mkdir(exist_ok=True)
         self.cache_dir.mkdir(parents=True, exist_ok=True)
+        source_date_epoch = self.config.environment.get("SOURCE_DATE_EPOCH")
+        self.source_date_epoch: Optional[int] = None
+        if source_date_epoch is not None:
```

Comment:
Drop `is not None` here. An empty setting of SOURCE_DATE_EPOCH= should be treated the same as not setting it at all.

Reply:
So we can check `len(source_date_epoch) > 0` instead?

Reply:
You *could*, but there is no reason to. The check `if source_date_epoch` is exactly what we need. It is equivalent to `if source_date_epoch is not None and source_date_epoch != ''`.